src/neuromancer/modules/blocks.py

```diff
@@ -982,6 +982,86 @@ def block_eval(self, x, hx=None):
         return self.output(hiddens[-1])
 
 
+class LSTM(Block):
+    def __init__(
+            self,
+            insize,
+            outsize,
+            hidden_size,
+            num_layers=1,
+            bias=True,
+            dropout=0.0
+    ):
+        """
+        insize (int): The size of each input feature vector.
+        outsize (int): The size of each output feature vector.
+        hidden_size (int): The number of features in the hidden state of each LSTM cell.
+        num_layers (int, optional): The number of recurrent layers. Defaults to 1.
+        bias (bool, optional): If `True`, adds a learnable bias to the output. Defaults to True.
+        dropout (float, optional): If non-zero, introduces a dropout layer on the outputs of each
+                                   LSTM layer except the last layer, with dropout probability equal
+                                   to the value specified. Defaults to 0.0.
+        """
+        super().__init__()
+        self.in_features, self.out_features = insize, outsize
+
+        self.lstm = torch.nn.LSTM(
+            input_size=insize,
+            hidden_size=hidden_size,
+            num_layers=num_layers,
+            bias=bias,
+            batch_first=True,
+            dropout=dropout,
+            bidirectional=False,
+            device=None,
+            dtype=None)
+
+        self.output = torch.nn.Linear(hidden_size, outsize)
+
+        self.h, self.c = None, None
+
+    def reset(self):
+        """
+        Resets the internal states (hidden state and cell state) of the LSTM module.
+        This should be called at the beginning of a sequence. You can use this with the neuromancer System class
+        by overriding the init function:
+
+        ssm = LSTM(args)
+        system_node = Node(ssm, ["yn", "U", "D"], ["yn"])
+        system = System([system_node])
+
+        def reset(self, data):
+            self.nodes[0].callable.reset() # reset the index of the node in the system that is the LSTM
+            return data
+        system.init = reset.__get__(system, System)
+        """
```

Comment:
Can this be done by passing the custom reset function as `init_func` when the `System` is constructed instead?